### DISCUSSION TITLE
Performance issue while connecting from linux container

### DISCUSSION THREAD

**Discussion by kalyan-raman** [dsbenghe/Novell.Directory.Ldap.NETStandard#184] [General]
Environment : Docker Desktop for Windows - Linux container
Novell.Directory.Ldap.NETStandard version: 3.6.0
.NET 5
Docker image - aspnet:5.0-1

I am trying to connect to ldap server over unsecured channel using SaslDigestMd5Request.  The connect and bind methods are taking over 2 seconds.

How to address this. Can the connection be registered as a singleton

**Comment by se** [comment]
Did you try to install your certificate in your Docker image?

**Comment by se** [comment]
@dsbenghe probably you need to close this 🖖

**Comment by dsbenghe** [comment]
You can debug these kind of issues i.e. find out if the performance issue is in the library or due to your environment by using one of the ldap tools i.e. on Linux ldap-utils from openldap - https://packages.ubuntu.com/bionic/ldap-utils
